tools/webhook-apicoverage/tools/tools.go

```diff
@@ -0,0 +1,141 @@
+	"os/user"
+	"path"
+
+	"github.com/knative/test-infra/tools/webhook-apicoverage/view"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	// Mysteriously required to support GCP auth (required by k8s libs).
+	// Apparently just importing it is enough. @_@ side effects @_@.
+	// https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// WebhookResourceCoverageEndPoint constant for resource coverage API endpoint.
+	WebhookResourceCoverageEndPoint = "https://%s:443/resourcecoverage?resource=%s"
+)
+
+// GetDefaultClusterName helper method to fetch cluster name from the env variable K8S_CLUSTER_OVERRIDE
+func GetDefaultClusterName() (string, error) {
+	envName := "K8S_CLUSTER_OVERRIDE"
+	var clusterName string
+	if clusterName = os.Getenv(envName); len(clusterName) == 0 {
+		return "", fmt.Errorf("cluster name not set. Method expects the cluster to be set as an environment variable: %s", envName)
+	}
+
+	return clusterName, nil
+}
+
+// GetDefaultKubePath helper method to fetch kubeconfig path.
+func GetDefaultKubePath() (string, error) {
+	var (
+		usr *user.User
+		err error
+	)
+	if usr, err = user.Current(); err != nil {
+		return "", fmt.Errorf("error retrieving current user: %v", err)
+	}
+
+	return path.Join(usr.HomeDir, ".kube/config"), nil
+}
+
+func getKubeClient(kubeConfigPath string, clusterName string) (*kubernetes.Clientset, error) {
+	overrides := clientcmd.ConfigOverrides{}
+	overrides.Context.Cluster = clusterName
+	clientCfg, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeConfigPath},
+		&overrides).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error building kube client config: %v", err)
+	}
+
+	var kubeClient *kubernetes.Clientset
+	if kubeClient, err = kubernetes.NewForConfig(clientCfg); err != nil {
+		return nil, fmt.Errorf("error building KubeClient from config: %v", err)
+	}
+
+	return kubeClient, nil
+}
+
+// GetWebhookServiceIP is a helper method to fetch IP Address of the LoadBalancer webhook service.
+func GetWebhookServiceIP(kubeConfigPath string, clusterName string, namespace string, serviceName string) (string, error) {
+	kubeClient, err := getKubeClient(kubeConfigPath, clusterName)
+	if err != nil {
+		return "", err
+	}
+
+	svc, err := kubeClient.CoreV1().Services(namespace).Get(serviceName, v1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error encountered while retrieving service: %s Error: %v", serviceName, err)
+	}
+
+	if len(svc.Status.LoadBalancer.Ingress) == 0 {
+		return "", fmt.Errorf("found zero Ingress instances for service: %s", serviceName)
+	}
+
+	return svc.Status.LoadBalancer.Ingress[0].IP, nil
+}
+
+// GetResourceCoverage is a helper method to get Coverage data for a resource from the service webhook.
+func GetResourceCoverage(webhookIP string, resourceName string) (string, error) {
```

Comment:
should we handle the case where service becomes unresponsive, set a timeout?

Reply:
This service is started by the webhook setup. We would not reach this stage if the service is not up because the tests would have failed by then. The timeout would be automatically setup by the httpClient.